packages/liveblocks-react/src/umbrella-store.ts

```diff
@@ -1025,6 +1033,57 @@ export class UmbrellaStore<M extends BaseMetadata> {
     this.setQuery2State(queryKey, { isLoading: false, error });
   }
 
+  public setLoadThreadsPromise(
+    queryKey: string,
+    getThreadsAndNotificationsPromise: Promise<{
+      threads: ThreadData<M>[];
+      inboxNotifications: InboxNotificationData[];
+    }>
+  ): Promise<ThreadData<M>[]> {
+    const loadThreadsPromise = autoRetry(
+      () => {
+        return getThreadsAndNotificationsPromise
+          .then((result) => {
+            window.console.log("DONE", result);
+            this.updateThreadsAndNotifications(
+              result.threads,
+              result.inboxNotifications,
+              [],
+              [],
+              queryKey
+            );
+            return result.threads;
+          })
+          .catch((err) => {
+            // Throw the error as we want `autoRetry` function to handle the error
+            throw err;
+          });
+      },
+      5,
+      [5000, 5000, 10000, 15000]
+    ).catch((err) => {
```

Comment:
We may have to set the promise status here (in the then and catch clauses) of autoRetry and after the request has been kicked off instead of doing it inside the use polyfill.

Reply:
Yess! Exactly my thinking, see comments above.